src/rplidar_node.cpp

```diff
@@ -157,36 +158,36 @@ void rplidar_node::publish_scan(
     scan_msg.angle_max = M_PI - angle_max;
   }
   scan_msg.angle_increment =
-    (scan_msg.angle_max - scan_msg.angle_min) / (double)(node_count);
+    (scan_msg.angle_max - scan_msg.angle_min) / (double)(node_count - 1);
 
   scan_msg.scan_time = scan_time;
-  scan_msg.time_increment = scan_time / (double)(node_count);
+  scan_msg.time_increment = scan_time / (double)(node_count - 1);
   scan_msg.range_min = min_distance;
   scan_msg.range_max = max_distance;
 
   scan_msg.intensities.resize(node_count);
   scan_msg.ranges.resize(node_count);
   bool reverse_data = (!inverted_ && reversed) || (inverted_ && !reversed);
-  if (!reverse_data) {
-    for (size_t i = 0; i < node_count; i++) {
+  size_t scan_midpoint = node_count / 2;
+  for (size_t i = 0; i < node_count; i++) {
```

Comment:
```suggestion
  for (size_t i = 0; i < node_count; ++i) {
```

Nit: I prefer pre-increment

Reply:
The diff is a little messy, but its just converting the 4 spaces to 2 and switching the increment you mentioned